tests/python/PyImportSortBearTest.py

```diff
@@ -21,3 +29,81 @@
      'import abc\nimport xyz\n'),
     settings={'known_standard_library_imports': 'xyz',
               'known_first_party_imports': 'abc'})
+
+
+class PyImportSortBearTest(unittest.TestCase):
+
+    def setUp(self):
+        self.queue = Queue()
+        self.section = Section('PyImportSortBear')
+        self.uut = PyImportSortBear(self.section,
+                                    self.queue)
+
+    def test_treat_seperated_imports_independently(self):
+        self.file = (
+            """
+import re
+import requests
+from urllib.parse import urlparse
+
+from difflib import SequenceMatcher
+from collections import defaultdict
+
+from coalib.results.Diff import Diff
+from coalib.bears.LocalBear import LocalBear
+from coalib.bears.requirements.PipRequirement import PipRequirement
+from coalib.results.RESULT_SEVERITY import RESULT_SEVERITY
+from coalib.results.Result import Result
+from coalib.bearlib import deprecate_settings
+""".splitlines(True)
+        )
+
+        expected = (
+            """
+import re
+from urllib.parse import urlparse
+
+import requests
+
+from collections import defaultdict
+from difflib import SequenceMatcher
+
+from coalib.bearlib import deprecate_settings
+from coalib.bears.LocalBear import LocalBear
+from coalib.bears.requirements.PipRequirement import PipRequirement
+from coalib.results.Diff import Diff
+from coalib.results.Result import Result
+from coalib.results.RESULT_SEVERITY import RESULT_SEVERITY
+""".splitlines(True)
+        )
+        diff = Diff(expected)
+        settings = {'treat_seperated_imports_independently': True}
+        self.assertEqual(list(self.uut.run('',
+                                           self.file,
+                                           **settings))[0].diffs[''].modified,
+                         diff.modified)
+
+        self.assertEqual(list(self.uut.run('',
+                                           ['import curses\n', 'import this\n',
+                                            'coala = "coala"\n'],
+                                           **settings)), [])
+
+    def test_import_seperation(self):
+        self.file = (
+            """
+from x import (y,
+               z,
+               w)
+import this
+
+import x, y
```

Comment:
`y` here shadows `y` in the first import statement.
pyflakes would scream about that.
maybe use `a, b` instead here.